pipelinewise/fastsync/commons/split_gzip.py

```diff
@@ -0,0 +1,127 @@
+import io
+import logging
+import gzip
+
+LOGGER = logging.getLogger(__name__)
+
+DEFAULT_CHUNK_SIZE_MB = 1000
+DEFAULT_MAX_CHUNKS = 20
+
+# Detecting compressed file size at write time is not possible by GzipFile.
+# The data hase to be written into the file first before the actual compression performed.
+# We need to use a good estimate for a text gzip file to split the file and write time.
+EST_COMPR_RATE = 0.12
+
+
+# pylint: disable=W0622
+def open(base_filename, mode='wb', chunk_size_mb=None, max_chunks=None, est_compr_rate=None):
+    """Open a gzip-compressed file in binary or text mode.
+
+    Args:
+        base_filename: Path where to create the zip file(s) with the exported data.
+                       Dynamic chunk numbers are appended to the base_filename
+        mode: "wb" or "wt". (Default: wb)
+        chunk_size_mb: File chunk sizes. (Default: 1000)
+        max_chunks: Max number of chunks. If set to 0 then splitting is disabled and one single
+                    file will be created (Default: 20)
+        est_compr_rate: Detecting compressed file size at write time is not possible by GzipFile.
+                        The data hase to be written into the file first before the actual compression performed.
+                        We need to use a good estimate for a text gzip file to split the file and write time.
+                        (Default 0.12)
+
+    Return:
+        File like object
+    """
+    if mode not in ['wb', 'wt']:
+        raise ValueError('Invalid mode: %r' % (mode,))
+    return SplitGzipFile(base_filename, mode, chunk_size_mb, max_chunks, est_compr_rate)
+
+
+# pylint: disable=R0902
+class SplitGzipFile(io.BufferedIOBase):
+    """The SplitGzipFile file like object class that implements only the write method.
+
+    This class only supports writing files in binary mode.
+    """
+    def __init__(self,
+                 base_filename,
+                 mode: str = None,
+                 chunk_size_mb: int = None,
+                 max_chunks: int = None,
+                 est_compr_rate: float = None):
+        super().__init__()
+
+        self.base_filename = base_filename
+        self.mode = mode
+        self.chunk_size_mb = chunk_size_mb if chunk_size_mb is not None else DEFAULT_CHUNK_SIZE_MB
+        self.max_chunks = max_chunks if max_chunks is not None else DEFAULT_MAX_CHUNKS
+        self.est_compr_rate = est_compr_rate if est_compr_rate is not None else EST_COMPR_RATE
+
+        self.split_large_files = True
+        self.chunk_seq = 1
+        self.current_chunk_size_mb = 0
+        self.chunk_filename = None
+        self.chunk_file = None
+
+    def _gen_chunk_filename(self) -> str:
+        """
+        Generates a chunk filename
+
+        Pattern if max_chunks is zero: <base_filename>
+        Pattern if max_chunks is greater than zero: <base_filename>.part<chunk-number-padded-five-digits>
+
+        Returns:
+            string
+        """
+        if self.max_chunks == 0:
```

Comment:
hmm, max_chunk min value is 1, why check for 0?

Reply:
`max_chunk == 0` means no split required and everything has to be written into one file, with no `.part000X` suffix:

```
        Pattern if max_chunks is zero: <base_filename>
        Pattern if max_chunks is greater than zero: <base_filename>.part<chunk-number-padded-five-digits>
```

Reply:
`split_file_max_chunks` min value is 1 but only in the PPW config.

split_gzip is separated object and it's ignoring splitting if `max_chunk` is zero.

The PPW config translated to split_gzip compatible parameter at
https://github.com/transferwise/pipelinewise/blob/9f6cbd87e83c0b82349f4e24b16407c044bbd579/pipelinewise/fastsync/commons/tap_postgres.py#L428

Reply:
> max_chunk == 0 means no split required and everything has to be written into one file

what about `max_chunk == 1` ?

Reply:
max_chunk 1 will create a file `abc.gzip.part.00001`